cmd/otel-allocator/allocation/least_weighted.go

```diff
@@ -0,0 +1,110 @@
+package allocation
```

Comment:
This should be own package along with the factory function.

Reply:
Agreed.  I was thinking about the visibility of some of the types and methods and was wondering whether the `Allocator` and the allocation strategy implementations shouldn't live in separate packages to ensure that they can be implemented without using any internals.  This would be important to allow someone to bring their own allocation strategy and plug it in.

Reply:
moving this to a new package means we lose the ability to use fields and types we don't exported like `State` and `collector`

Reply:
`State` is exported, `collector` is not.  Would it make sense to export `Collector`?  I think so.